wiki/CONTRIBUTING.md

```diff
@@ -0,0 +1,141 @@
+or you can run `npm run build` from `automation/Extension/firefox/`.
+
+### Debugging the platform
+
+Manual debugging with OpenWPM can be difficult. By design the platform runs all
+browsers in separate processes and swallows all exceptions (with the intent of
+continuing the crawl). We recommend using
+[manual_test.py](../test/manual_test.py).
+
+This utility allows manual debugging of the extension instrumentation with or
+without Selenium enabled, as well as makes it easy to launch a Selenium
+instance (without any instrumentation)
+* `./scripts/build-extension.sh`
+* `python -m test.manual_test` builds the current extension directory
+  and launches a Firefox instance with it.
+* `python -m test.manual_test --selenium` launches a Firefox Selenium instance
+  after automatically rebuilding `openwpm.xpi`. The script then
+  drops into an `ipython` shell where the webdriver instance is available
+  through variable `driver`.
+* `python -m test.manual_test --selenium --no_extension` launches a Firefox Selenium
+  instance with no instrumentation. The script then
+  drops into an `ipython` shell where the webdriver instance is available
+  through variable `driver`.
+
+### Managing requirements
+
+We use a script to pin dependencies `scripts/repin.sh`.
+
+This means that `environment.yaml` should not be edited directly.
+
+Instead, place new requirements in `scripts/environment-unpinned.yaml` or `scripts/environment-unpinned-dev.yaml`
+and then run repin:
+
+    $ cd scripts
+    $ ./repin.sh
+
+To update the version of firefox, the TAG variable must be updated in the `./scripts/install-firefox.sh`
+script. This script contains further information about finding the right TAG.
+
+### Running tests
+
+OpenWPM's tests are build on [pytest](https://docs.pytest.org/en/latest/). Execute `py.test -vv`
+in the test directory to run all tests:
+
+    $ conda activate openwpm
+    $ cd test
+    $ py.test -vv
+
+See the [pytest docs](https://docs.pytest.org/en/latest/) for more information on selecting
+specific tests and various pytest options.
+
+### Mac OSX
+
+You may need to install `make` / `gcc` in order to build the extension.
+The necessary packages are part of xcode: `xcode-select --install`
+
+We do not run CI tests for Mac, so new issues may arise. We welcome PRs to fix
+these issues and add full CI testing for Mac.
+
+Running Firefox with xvfb on OSX is untested and will require the user to install
+an X11 server. We suggest [XQuartz](https://www.xquartz.org/). This setup has not
+been tested, we welcome feedback as to whether this is working.
+
+### Updating schema docs
+
+In the rare instance that you need to create schema docs
+(after updating or adding files to `schemas` folder), run `npm install`
+from OpenWPM top level. Then run `npm run render_schema_docs`. This will update the
+`docs/schemas` folder. You may want to clean out the `docs/schemas` folder before doing this
+incase files have been renamed.
+
+## Troubleshooting
```

Comment:
All of these issues may hit a person just trying to install and run OpenWPM? They should be moved back into the main README. I think we can merge them with the other troubleshooting section.